addons/web/static/src/webclient/webclient.js

```diff
@@ -149,6 +150,15 @@ export class WebClient extends Component {
         if (navigator.serviceWorker) {
             navigator.serviceWorker
                 .register("/web/service-worker.js", { scope: "/odoo" })
+                .then(() => {
+                    navigator.serviceWorker.ready.then(() => {
+                        if (!navigator.serviceWorker.controller) {
+                            console.log("hard reload => invalidate caches");
+                            rpcBus.trigger("CLEAR-CACHES");
+                            // https://stackoverflow.com/questions/51597231/register-service-worker-after-hard-refresh
+                        }
```

Comment:
I would put this in its own commit
It's really experimental, a hackish heuristic that I'm not sure we'll keep